src/validation.cpp

```diff
@@ -1931,7 +1931,9 @@ bool static DisconnectTip(CValidationState& state, const CChainParams& chainpara
             disconnectpool->removeEntry(it);
         }
     }
+
     // Update MN manager cache
+    deterministicMNManager->UpdatedBlockTip(pindexDelete->pprev);
```

Comment:
In a022354d:

Might be better to move this call below the `UpdateTip(pindexDelete->pprev);` at line 1949. Otherwise if for some reason the manager access the chain interface, it will have a different tip.

(Same applies to the `mnodeman.cacheBlockHash` call that is below actually).

Reply:
:+1: done